packages/shared-react/src/api/persistQueryClientContext.tsx

```diff
@@ -45,37 +53,43 @@ export const PersistQueryClientContextProvider = ({ children }: { children: Reac
     queryClient.clear()
   }, [unsubscribeFromQueryClientFn])
 
-  const restoreQueryCache = useCallback(async (walletId: string, isPassphraseUsed?: boolean) => {
-    setIsRestoring(true)
-
-    if (!isPassphraseUsed) {
-      const options: PersistQueryClientOptions = {
-        queryClient,
-        maxAge: Infinity,
-        persister: createTanstackIndexedDBPersister('tanstack-cache-for-wallet-' + walletId),
-        dehydrateOptions: {
-          shouldDehydrateQuery: (query) =>
-            query.meta?.['isMainnet'] === false ? false : defaultShouldDehydrateQuery(query)
+  const restoreQueryCache = useCallback(
+    async (walletId: string, isPassphraseUsed?: boolean) => {
+      setIsRestoring(true)
+
+      if (!isPassphraseUsed) {
+        const options: PersistQueryClientOptions = {
+          queryClient,
+          maxAge: Infinity,
+          persister: createPersister('tanstack-cache-for-wallet-' + walletId),
+          dehydrateOptions: {
+            shouldDehydrateQuery: (query) =>
+              query.meta?.['isMainnet'] === false ? false : defaultShouldDehydrateQuery(query)
+          }
         }
-      }
 
-      await persistQueryClientRestore(options)
+        await persistQueryClientRestore(options)
 
-      const newUnsubscribeFromQueryClientFn = persistQueryClientSubscribe(options)
+        const newUnsubscribeFromQueryClientFn = persistQueryClientSubscribe(options)
 
-      setUnsubscribeFromQueryClientFn(() => newUnsubscribeFromQueryClientFn)
-    } else {
-      // Even when we don't restore data in the case of passphrase wallet, we need to set `isRestoring` to `true` and
-      // then to `false` to make sure the useQuery instances are reset.
-      await sleep(500)
-    }
+        setUnsubscribeFromQueryClientFn(() => newUnsubscribeFromQueryClientFn)
+      } else {
+        // Even when we don't restore data in the case of passphrase wallet, we need to set `isRestoring` to `true` and
+        // then to `false` to make sure the useQuery instances are reset.
+        await sleep(500)
+      }
 
-    setIsRestoring(false)
-  }, [])
+      setIsRestoring(false)
+    },
+    [createPersister]
+  )
 
-  const deletePersistedCache = useCallback((walletId: string) => {
-    createTanstackIndexedDBPersister('tanstack-cache-for-wallet-' + walletId).removeClient()
-  }, [])
+  const deletePersistedCache = useCallback(
+    (walletId: string) => {
+      createPersister('tanstack-cache-for-wallet-' + walletId).removeClient()
```

Comment:
Nit: It's a bit weird to me that we run the create function again to simply remove the client. But I don't mind too much :P